eppo_client/client.py

```diff
@@ -183,114 +184,59 @@ def get_assignment_variation(
         """
         validate_not_blank("subject_key", subject_key)
         validate_not_blank("flag_key", flag_key)
-        experiment_config = self.__config_requestor.get_configuration(flag_key)
-        override = self._get_subject_variation_override(experiment_config, subject_key)
-        if override:
-            if expected_variation_type is not None:
-                variation_is_expected_type = VariationType.is_expected_type(
-                    override, expected_variation_type
-                )
-                if not variation_is_expected_type:
-                    return None
-            return override
+        if not subject_attributes:
```

Comment:
Super-nit: prefer `if subject_attributes is None`. Given the type restriction, `None` is the only falsy value (other than `{}`), so this would just be to make it more standard/readable without ever changing behavior.